build.gradle

```diff
@@ -107,6 +119,29 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
+String checkForNewEap (String gradleMajor){
+  Document htmlContent = Jsoup.connect("https://www.jetbrains.com/intellij-repository/snapshots/").get()
+  String link = htmlContent.select('a[href^=\"https://www.jetbrains.com/intellij-repository/snapshots/com/jetbrains/intellij/idea/ideaIC/'+gradleMajor+'.\"][href$="-EAP-SNAPSHOT.pom"]')[0].attr("href")
+  Pattern pattern = Pattern.compile("ideaIC-(.*?).pom")
+  Matcher matcher = pattern.matcher(link)
+  matcher.find()
+  return matcher.group(1)
+}
+
+task updateEapOfLatestSupportedMajor {
+  Properties properties = intellijVersionsProperties()
+  File propFile = intellijVersionsFile()
+  properties.load(propFile.newDataInputStream())
+  String latest = properties.getProperty("eapOfLatestSupportedMajor")
+  if (latest){
+    String newerVersion = project.checkForNewEap(latest.split("\\.")[0])
+    if(!newerVersion.equals(latest)){
+      properties.setProperty("eapOfLatestSupportedMajor", newerVersion)
+      properties.store(propFile.newWriter(), null)
+    }
+  }
```

Comment:
The missing `else` here is also pretty important! Coz as for now the task will only be able to bump EAP version if some is provided, but not able to detect a new EAP :/

Reply:
but when eapOfLatestSupportedMajor is null (or empty), what is the meaning of the new EAP ? just the latest available ?

Reply:
> but when eapOfLatestSupportedMajor is null

This is the case right now, as of May 16th: we've got 2022.1 released quite recently, but an `EAP-SNAPSHOT` for the next version, **2022.2**, has NOT been published yet.

Buuuuuut... there's some foreshadowing! Check https://www.jetbrains.com/intellij-repository/snapshots/ -> Ctrl+F .idea... there's already `222.2270-EAP-CANDIDATE-SNAPSHOT` from literally yesterday. The leading `222` is shortened notation for 20**22.2** (same for all other build ids, e.g. 193 was 2019.3 etc.).